docs/src/proposals/timely-vote-credits.md

```diff
@@ -0,0 +1,144 @@
+for distant validators who are not artificially lagging.
+
+Historical voting data has been analyzed over many epochs and the data
+suggests that the smallest grace period that allows for very minimal impact on
+well behaved distant validators is 3 slots, which means that all slots voted
+on within 3 slots will award maximum vote credits to the voting validator.
+This gives validators nearly 2 seconds to land their votes without penalty.
+The minimum latency between two points on Earth is about 100 ms, so allowing a
+full 1,500 ms to 2,000 ms latency without penalty should not have adverse
+impact on distant validators.
+
+### Maximum Vote Credits
+
+Another factor to consider is what the maximum vote credits to award for a
+vote should be.  Assuming linear reduction in vote credits awarded (where 1
+slot of additional lag reduces earned vote credits by 1), the maximum vote
+credits value determines how much "penalty" there is for each additional slot
+of latency.  For example, a value of 10 would mean that after the grace period
+slots, every additional slot of latency would result in a 10% reduction in
+vote credits earned as each subsequent slot earns 1 credit less out of a
+maximum possible 10 credits.
+
+Again, historical voting data was analyzed over many epochs and the conclusion
+drawn was that a maximum credits of 10 is the largest value that can be used
+and still have a noticeable effect on known laggers.  Values higher than that
+result in such a small penalty for each slot of lagging that intentional
+lagging is still too profitable.  Lower values are even more punishing to
+intentional lagging; but an attempt has been made to conservatively choose the
+highest value that produces noticeable results.
+
+The selection of these values is partially documented here:
+
+https://www.shinobi-systems.com/timely_voting_proposal
+
+The above document is somewhat out of date with more recent analysis, which
+occurred in this github issue:
+
+https://github.com/solana-labs/solana/issues/19002
+
+To summarize the findings of these documents: analysis over many epochs showed
+that almost all validators from all regions have an average vote latency of 1
+slot or less.  The validators with higher average latency are either known
+laggers, or are not representative of their region since many other validators
+in the same region achieve low latency voting.  With a maximum vote credit of
+10, there is almost no change in vote credits earned relative to the highest vote
+earner by the majority of validators, aside from a general uplift of about 0.4%.
+Additionally, data centers were analyzed to ensure that there aren't regions of
+the world that would be adversely affected, and none were found.
+
+
+### Method of Implementation
+
+When a Vote or VoteStateUpdate instruction is received by a validator, it will
+use the Clock sysvar to identify the slot in which that instruction has
+landed.  For any newly voted on slot within that Vote or VoteStateUpdate
```

Comment:
You probably mentioned this elsewhere but remind me again: how does this get handled for VoteStateUpdate's which might not necessarily have a vote for every slot? also with the recent changes we ignore redundant VoteStateUpdate's in gossip and tpu for faster processing which I assume will be a factor as well...

Reply:
awesome, might be useful to put (1) (2) and (3) in the proposal. 
I think you could also mention something about how optimizations might coalesce certain VoteStateUpdate's together but the latency based rewards should be enough to deter a validator from submitting "incremental" VoteStateUpdates